python/pyarrow/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -656,6 +693,33 @@ Status AppendUTF32(const char* data, int itemsize, int byteorder,
       reinterpret_cast<const uint8_t*>(PyBytes_AS_STRING(utf8_obj.obj())), length);
 }
 
+Status AppendUTF32(const char* data, int itemsize, int byteorder,
+                   ::arrow::LargeStringBuilder* builder) {
```

Comment:
For this function, only the final line of the function actually uses `builder`. So I think it should be possible to factor out most part of this function into something that can be reused, and only have the actual `builder->Append(..)` in a type-specific AppendUTF32?

Reply:
Figured out how to get it working as a template, wasn't expecting it to be possible without a common base class for the two builders